### PR TITLE
Build paypal with sylius 1.8 and 1.9

### PR DESCRIPTION
**Pull request by SirDomin**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -13,20 +13,24 @@ jobs:
...
+        name: "PHP ${{ matrix.php }}, MySQL ${{ matrix.mysql }}, Sylius ${{ matrix.sylius }}"
 
         strategy:
             fail-fast: false
             matrix:
                 php: [7.4, 7.3]
                 node: [10.x]
                 mysql: [5.7, 8.0]
+                symfony: [^4.4, ^5.2]
+                sylius: [~1.8.0, ~1.9.0@beta]
```
Comment (Zales0123) [inline .github/workflows/build.yml:25]: Why do we use 1.9@beta when 1.9 stable is already available?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -83,6 +87,23 @@ jobs:
                     restore-keys: |
                         ${{ runner.os }}-php-${{ matrix.php }}-composer-
 
+            -
+                name: Install Sylius version
```
Comment (Zales0123) [inline .github/workflows/build.yml:91]: We probably don't need this step as we have `Restrict Sylius version` below

File: src/Onboarding/Processor/BasicOnboardingProcessor.php [hunk 3]
```diff
@@ -76,7 +76,7 @@ public function process(
             'partner_attribution_id' => $response['partner_attribution_id'],
         ]);
 
-        $permissionsGranted = (bool) $request->query->get('permissionsGranted', true);
+        $permissionsGranted = $request->query->get('permissionsGranted') === null ? true : (bool) $request->query->get('permissionsGranted');
```
Comment (Zales0123) [inline src/Onboarding/Processor/BasicOnboardingProcessor.php:79]: This looks strange, the previous implementation should work fine (we have `true` as a default value). Psalm couldn't handle it?

File: tests/Application/Kernel.php [hunk 4]
```diff
@@ -103,4 +99,41 @@ private function isTestEnvironment(): bool
...
+    private function loadRoutesConfiguration(RouteCollectionBuilder $routes, string $confDir): void
+    {
+        $routes->import($confDir . '/{routes}/*' . self::CONFIG_EXTS, '/', 'glob');
+        $routes->import($confDir . '/{routes}/' . $this->environment . '/**/*' . self::CONFIG_EXTS, '/', 'glob');
+        $routes->import($confDir . '/{routes}' . self::CONFIG_EXTS, '/', 'glob');
+    }
+
+    /**
+     * @return BundleInterface[]
+     */
```
Comment (Zales0123) [inline tests/Application/Kernel.php:120]: Could be shorter:

```php
/** @return BundleInterface[] */
```

same below

File: tests/Application/Kernel.php [hunk 5]
```diff
@@ -103,4 +99,41 @@ private function isTestEnvironment(): bool
...
+    private function registerBundlesFromFile(string $bundlesFile): iterable
+    {
+        $contents = require $bundlesFile;
+        foreach ($contents as $class => $envs) {
```
Comment (Zales0123) [inline tests/Application/Kernel.php:124]: `$envs` => `$environments`? #abbreviationsAreEvil

**Review by Zales0123 — CHANGES_REQUESTED** [review]

File: src/Onboarding/Processor/BasicOnboardingProcessor.php (hunk 3 above)
Comment (SirDomin) [reply src/Onboarding/Processor/BasicOnboardingProcessor.php:79]: in newer versions second parameter must be null || string so thats the way around it :/

File: composer.json [hunk 6]
```diff
@@ -7,7 +7,7 @@
     "require": {
         "php": "^7.3",
 
-        "sylius/sylius": "1.8.*",
+        "sylius/sylius": "~1.8.0 || ~1.9.0@beta",
```
Comment (Zales0123) [inline composer.json:10]: Isn't `^1.8` enough?

**Review by Zales0123 — APPROVED** [review]

**Pull request merged**